File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53404/output/project/src/main/java/customerimporter/DNI.java
package customerimporter;

import static customerimporter.Validation.assertThat;
import static customerimporter.Validation.isAllDigits;
import static customerimporter.Validation.isBetween;

import java.util.function.Supplier;

public class DNI extends Identification {

    public static final String TYPE_CODE = "D";

    private static final int MINIMUM_NUMBER = 1;
    private static final int MAXIMUM_NUMBER = 99999999;

    // instance creation

    public static DNI with(String aNumber) {
        assertIsValidNumber(aNumber);
        return new DNI(aNumber);
    }

    // error messages

    public static String invalidNumberErrorDescription() {
        return "Invalid DNI number";
    }

    // asserting - private

    private static void assertIsValidNumber(String aNumber) {
        assertThat(isAllDigits(aNumber), invalidNumberErrorDescription());
        assertThat(isBetween(Integer.parseInt(aNumber), MINIMUM_NUMBER, MAXIMUM_NUMBER), invalidNumberErrorDescription());
    }

    // initialization

    private DNI(String aNumber) {
        super(aNumber);
    }

    // accessing

    public String typeCode() {
        return TYPE_CODE;
    }

    // testing

    public boolean isCUIT() {
        return false;
    }

    public boolean isDNI() {
        return true;
    }

    // evaluating

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return Integer.parseInt(number());
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-211617-6449/analysis/acceptance-project/src/customerimporter/dni.py
from customerimporter.identification import Identification


class Dni(Identification):

    # type

    @classmethod
    def type_code(cls):
        return "D"

    # validation

    @classmethod
    def assert_valid(cls, a_number):
        cls._assert_number_between(a_number, cls.minimum_number(), cls.maximum_number())

    @classmethod
    def maximum_number(cls):
        return 99999999

    @classmethod
    def minimum_number(cls):
        return 1

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid DNI number"

    # number

    def dni_number_if_none(self, a_none_block):
        return int(self.number())

    # testing

    def is_dni(self):
        return True
